Randomizing default helm credentials

Param-29 opened the pull request:
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

Param-29 committed:
secrets to be used instead of default values

Param-29 committed:
Support for all mongo secrets

rasswanth-s committed:
removed extra comments in mongo statefulset

rasswanth-s committed:
added secrets to the templates file for debugging
fixed error with mongosecret naming

madhavajay commented:
Suggestions:
- change {{}} go templates to '{{}}' syntax in .yaml file
- remove /helm chart folder from pre-commit yaml
- add lookup syntax to secrets so that it keeps existing secrets

Param-29 committed:
Merge branch 'dev' into paramm/secrets-file

Param-29 committed:
moved secrets.txt from secrets.yaml

Param-29 committed:
removing secrets.yaml

rasswanth-s committed:
Merge branch 'dev' into paramm/secrets-file

Param-29 committed:
lookup added for 4 secrets in mongodb

Param-29 committed:
precommit fix

rasswanth-s committed:
base64 fix for default root email and password

rasswanth-s commented on packages/grid/k8s/manifests/secrets.txt:
```diff
@@ -0,0 +1,39 @@
+  {{- $secretData := (get $secretObj "data") | default dict }}
+  # set $jwtSecret to existing secret data or generate a random one when not exists
+
+  # Sample below
```
NIT: we could remove them

rasswanth-s commented on packages/grid/k8s/manifests/secrets.txt:
```diff
@@ -0,0 +1,39 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "mongosecret"
```
We could also add a syft secret for randomizing syft specific secrets

Param-29 committed:
Merge branch 'dev' into paramm/secrets-file

Param-29 committed:
created 2 seperate secrets:
one for mongo, other for syft

Param-29 commented:
from local testing
```
(syft311) $ kubectl get secrets -n syft
NAME                          TYPE                 DATA   AGE
mongosecret                   Opaque               4      94s
syftsecret                    Opaque               2      94s
sh.helm.release.v1.syft3.v1   helm.sh/release.v1   1      94s
$ kubectl describe secret mongosecret -n syft
Name:         mongosecret
Namespace:    syft
Labels:       app.kubernetes.io/managed-by=Helm
Annotations:  meta.helm.sh/release-name: syft3
              meta.helm.sh/release-namespace: syft

Type:  Opaque

Data
====
mongo-password-secret:    32 bytes
mongo-username-secret:    32 bytes
mongoInitdbRootPassword:  32 bytes
mongoInitdbRootUsername:  32 bytes
$ kubectl describe secret syftsecret -n 
syft
Name:         syftsecret
Namespace:    syft
Labels:       app.kubernetes.io/managed-by=Helm
Annotations:  meta.helm.sh/release-name: syft3
              meta.helm.sh/release-namespace: syft

Type:  Opaque

Data
====
defaultRootPassword:  32 bytes
stackApiKey:          32 bytes
```

Param-29 committed:
Merge branch 'dev' into paramm/secrets-file

Param-29 committed:
precommit fix

Param-29 committed:
Merge branch 'dev' into paramm/secrets-file

Param-29 commented:
```
(syft311) param@param-Legion-5-15IAH7H:~/projects/oss/PySyft/packages/grid$ kubectl describe secrets seaweedfsecret -n syft4
Name:         seaweedfsecret
Namespace:    syft4
Labels:       app.kubernetes.io/managed-by=Helm
Annotations:  meta.helm.sh/release-name: syft4
              meta.helm.sh/release-namespace: syft4

Type:  Opaque

Data
====
S3_ROOT_PWD_SECRET:   32 bytes
S3_ROOT_USER_SECRET:  32 bytes
```

Param-29 committed:
seaweedfs randomizing

rasswanth-s commented on packages/grid/helm/helm.py:
```diff
@@ -105,6 +105,23 @@ def ingress_with_tls() -> str:
+    syft_secrets_source = os.path.join("k8s/manifests", "syft_secrets.txt")
+    with open(syft_secrets_path, "w") as fp:
+        with open(syft_secrets_source) as fp2:
+            fp.write(fp2.read())
```
Do we need to add seaweedfs_secrets.txt?

rasswanth-s commented on packages/grid/helm/syft/values.yaml:
```diff
@@ -4,22 +4,37 @@
       mongoHost: "mongo"
 
+seaweedfs:
+  start_variables:
```
MAYBE: we could remove this part?

rasswanth-s commented on packages/grid/helm/syft/values.yaml:
```diff
@@ -4,22 +4,37 @@
+      # mongoInitdbRootUsername: "root"
+      mongosecret: "mongosecret"
+  seaweedfs:
+    start_vars:
```
We could also remove start_vars in this particular part

Param-29 committed:
seeweedfs changes v2, start_vars removed

rasswanth-s commented on packages/grid/k8s/manifests/syft_secrets.txt:
```diff
@@ -0,0 +1,21 @@
+  {{- $secretObj := (lookup "v1" "Secret" .Release.Namespace "syftsecret") | default dict }}
+  {{- $secretData := (get $secretObj "data") | default dict }}
+
+  {{- $stackApiKey := (get $secretData "mongo-username-secret") | default (randAlphaNum 32 | b64enc) }}
```
I think the secretdata is stackApiKey instead of `mongo-username-secret`

rasswanth-s commented on packages/grid/k8s/manifests/syft_secrets.txt:
```diff
@@ -0,0 +1,21 @@
+  {{- $stackApiKey := (get $secretData "mongo-username-secret") | default (randAlphaNum 32 | b64enc) }}
+  stackApiKey: {{ $stackApiKey | quote }}
+
+  {{- $defaultRootPassword := (get $secretData "mongo-password-secret") | default (randAlphaNum 32 | b64enc) }}
```
I think the secretdata is defaultRootPassword instead of mongo-password-secret

rasswanth-s commented on packages/grid/k8s/manifests/mongo_secrets.txt:
```diff
@@ -0,0 +1,40 @@
+  {{- $mongoDbRootPass := (get $secretData "mongoInitdbRootPassword") | default (randAlphaNum 32 | b64enc) }}
+  mongoInitdbRootPassword: {{ $mongoDbRootPass | quote }}
+
+  # mongo-username-secret: {{ randAlphaNum 32 | b64enc | quote }}
```
MAYBE: we could remove the unused comments

Param-29 committed:
Fix for Syft_secrets + cleanup

rasswanth-s committed:
updated helm template files

rasswanth-s committed:
updated helm charts for configuring registry and syft version

rasswanth-s committed:
added registry and version to devspace

rasswanth-s committed:
fixed tag setting in devspace file

rasswanth-s committed:
remove unsed variable names

rasswanth-s committed:
added node name attribute in devspace

rasswanth-s committed:
fixed gateway profile in devspace

rasswanth-s committed:
fixed devspace dev mode

rasswanth-s committed:
Merge branch 'dev' into rasswanth/shift_devspace_to_helm

rasswanth-s committed:
addded helm syft version to bump2version and cd pipeline

rasswanth-s committed:
shifted custom notes generation function

rasswanth-s committed:
removed unused packages

rasswanth-s committed:
updated syft version

rasswanth-s committed:
removed tag during build images

rasswanth-s committed:
trying removing component-chart

rasswanth-s committed:
replace buildjet cache with actions/cache

rasswanth-s committed:
removed component chart and updated helm notes.txt

rasswanth-s committed:
added helm templates to cd-pipeline

rasswanth-s committed:
Merge branch 'dev' into rasswanth/shift_devspace_to_helm

rasswanth-s committed:
Merge branch 'dev' into paramm/secrets-file

rasswanth-s committed:
Merge branch 'rasswanth/shift_devspace_to_helm' into paramm/secrets-file

rasswanth-s committed:
- Moved the secrets randomization to syft templates folder

Co-authored-by: Param-29 <param.mirani1999@gmail.com>

rasswanth-s committed:
fixed helm string requirement

rasswanth-s committed:
fix syntax error in helm templates for referencing secrets

rasswanth-s committed:
simplified base64 conversion automatically during secrets creation

rasswanth-s committed:
- Moved Non-Public Information to node settings
- Deduplicated values in mongo-secret

rasswanth-s committed:
added devmode to devspace configuration

rasswanth-s committed:
removed unsed secrets text file

rasswanth-s committed:
- Added seaweedfs specific arguments to backend statefulset
- Removed seaweedfs username during randomization

rasswanth-s committed:
- Removed mongo user name from randomization
- Added a new field for mongo username in values.yaml

rasswanth-s committed:
added retrieval of password by environment variables for login

rasswanth-s committed:
removed unused fields in kubernetes tox test

rasswanth-s committed:
- Combined Variables names in tox
- As environment variables cannot have hypens
- And cluster names cannot have underscores
- These are required for passing the password as environment variables to test randomization

rasswanth-s committed:
-  Fixed Node Name intergration test
- Add environment variable mapping for randomized password in tox

rasswanth-s committed:
fixed randomized testing on CI

rasswanth-s committed:
removed unused k8s files

rasswanth-s committed:
Merge branch 'dev' into paramm/secrets-file

rasswanth-s committed:
updated values.yaml to new syft beta version

rasswanth-s reviewed (approved):
Great work @Param-29  🚀  🎸
